akka-stream/src/main/scala/akka/stream/impl/CompletedPublishers.scala

```diff
@@ -24,9 +29,31 @@ private[akka] case object EmptyPublisher extends Publisher[Nothing] {
 private[akka] final case class ErrorPublisher(t: Throwable, name: String) extends Publisher[Nothing] {
   import ReactiveStreamsCompliance._
   override def subscribe(subscriber: Subscriber[_ >: Nothing]): Unit =
-    try tryOnError(subscriber, t) catch {
+    try {
+      requireNonNullSubscriber(subscriber)
+      tryOnSubscribe(subscriber, CancelledSubscription)
+      tryOnError(subscriber, t)
+    } catch {
       case _: SpecViolation ⇒ // nothing we can do
     }
   def apply[T]: Publisher[T] = this.asInstanceOf[Publisher[T]]
   override def toString: String = name
 }
+
+/**
+ * INTERNAL API
```

Comment:
please add a comment that this is only legal if used to onSubscribe with an immediately following termination signal